lib/rust/lexer/definition/src/token.rs

```diff
@@ -216,6 +324,124 @@ impl Default for LineEnding {
 
 
 
+// =================
+// === TextStyle ===
+// =================
+
+/// The style of the text literal.
+#[derive(Copy,Clone,Debug,Eq,PartialEq)]
+pub enum TextStyle {
+    // === Line ===
+
+    /// A interpolated text line literal.
+    FormatLine,
+    /// A raw text line literal.
+    RawLine,
+    /// An unclosed text line literal.
+    UnclosedLine,
+
+    // === Inline Block ===
+
+    /// A format inline block text literal.
+    FormatInlineBlock,
+    /// A raw inline block text literal.
+    RawInlineBlock,
+
+    // === Block ===
+
+    /// An interpolated text block literal.
+    FormatBlock,
+    /// A raw text block literal.
+    RawBlock,
+}
+
+impl TextStyle {
+    /// Calculate the length of the delimiters for a particular style of text literal.
+    pub fn length(self) -> usize {
+        match self {
+            TextStyle::FormatLine        => 2,
+            TextStyle::RawLine           => 2,
+            TextStyle::FormatInlineBlock => 3,
+            TextStyle::RawInlineBlock    => 3,
+            TextStyle::UnclosedLine      => 1,
+            TextStyle::FormatBlock       => 3,
+            TextStyle::RawBlock          => 3,
```

Comment:
again, we have a big magic here. What if we change the lexing rules? This needs to be defined as a single constant and computed automatically,.